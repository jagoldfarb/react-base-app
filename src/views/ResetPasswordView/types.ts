export interface IResetPasswordFormData {
  password: string;
  passwordConfirmation: string;
}

export interface IResetPasswordViewProps {
  onSubmit: (formData: IResetPasswordFormData) => void;
}
