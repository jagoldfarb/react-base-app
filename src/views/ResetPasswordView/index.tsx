import React, { FunctionComponent } from 'react';
import i18next from 'i18next';

import ResetPasswordForm from 'views/ResetPasswordView/components/ResetPasswordForm';

import { IResetPasswordViewProps } from './types';
import {
  ResetPasswordContainer,
  ResetPasswordContent,
  Title,
  Description,
  ResetPasswordFormContainer,
  CreateAccountContainer,
  CreateAccountLink,
} from './styles';
import './i18n';

const ResetPasswordView: FunctionComponent<IResetPasswordViewProps> = (
  props: IResetPasswordViewProps,
) => {
  const { onSubmit } = props;

  return (
    <ResetPasswordContainer data-testid="reset-password-view">
      <ResetPasswordContent>
        <Title>{i18next.t('RESET_PASSWORD_VIEW:TITLE')}</Title>
        <Description>{i18next.t('RESET_PASSWORD_VIEW:DESCRIPTION')}</Description>
        <ResetPasswordFormContainer>
          <ResetPasswordForm onSubmit={onSubmit} />
        </ResetPasswordFormContainer>
        <CreateAccountContainer>
          <CreateAccountLink to="/signup">
            {i18next.t('RESET_PASSWORD_VIEW:CREATE_ACCOUNT')}
          </CreateAccountLink>
        </CreateAccountContainer>
      </ResetPasswordContent>
    </ResetPasswordContainer>
  );
};

export default ResetPasswordView;
