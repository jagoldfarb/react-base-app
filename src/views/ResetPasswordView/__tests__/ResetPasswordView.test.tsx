import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import store from 'redux/store';

import ResetPasswordView from '..';

const defaultProps = {
  onSubmit: jest.fn(),
};

describe('ResetPassword View', () => {
  it('Should render the reset password page', () => {
    const { asFragment, getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter>
          <ResetPasswordView {...defaultProps} />
        </MemoryRouter>
      </Provider>,
    );
    expect(getByTestId('reset-password-view')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
