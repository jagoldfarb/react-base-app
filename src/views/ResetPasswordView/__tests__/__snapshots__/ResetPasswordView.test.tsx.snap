// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword View Should render the reset password page 1`] = `
<DocumentFragment>
  <div
    class="sc-iBPRYJ jfuLoK"
    data-testid="reset-password-view"
  >
    <div
      class="sc-fubCfw ljPXGJ"
    >
      <h1
        class="sc-pFZIQ fbkcvO"
      >
        Reset your password
      </h1>
      <span
        class="sc-jrAGrp bsjQrB"
      >
        Enter your new password for your account.
      </span>
      <div
        class="sc-kEjbxe kdFnRR"
      >
        <form
          class="sc-eCssSg jvctCF"
          data-testid="reset-password-form"
        >
          <div
            class="sc-jSgupP bDwpav"
          >
            <div
              class="sc-bdfBwQ cgTScR"
            >
              <div
                class="MuiFormControl-root MuiTextField-root sc-gsTCUz jKtOdX"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                >
                  Password
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    data-testid="input-Password"
                    maxlength="255"
                    name="password"
                    rows="3"
                    type="password"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <button
                      aria-label="toggle password visibility"
                      class="MuiButtonBase-root MuiIconButton-root"
                      data-testid="toggle-password-button"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-3"
                    >
                      <span>
                        Password
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-jSgupP bDwpav"
          >
            <div
              class="sc-bdfBwQ cgTScR"
            >
              <div
                class="MuiFormControl-root MuiTextField-root sc-gsTCUz jKtOdX"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                >
                  Password confirmation
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    data-testid="input-Password confirmation"
                    maxlength="255"
                    name="passwordConfirmation"
                    rows="3"
                    type="password"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <button
                      aria-label="toggle password visibility"
                      class="MuiButtonBase-root MuiIconButton-root"
                      data-testid="toggle-password-button"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-3"
                    >
                      <span>
                        Password confirmation
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-gKsewC fiOhoO"
          >
            <button
              class="sc-hKgILt bytOKm"
              data-testid="button"
              type="submit"
            >
              Change password
            </button>
          </div>
        </form>
      </div>
      <div
        class="sc-iqHYGH loMoME"
      >
        <a
          class="sc-crrsfI fpvyaq"
          href="/signup"
        >
          Create account
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
